watertap/core/zero_order_diso.py

```diff
@@ -0,0 +1,244 @@
+                b.properties_treated[t].flow_mass_comp["H2O"])
+
+    # Solute concentration of treated stream
+    @self.Constraint(self.flowsheet().time,
+                     self.config.property_package.solute_set,
+                     doc='Constraint for solute concentration in treated '
+                         'stream.')
+    def solute_treated_equation(b, t, j):
+        return ((1 - b.removal_frac_mass_solute[t, j]) *
+                (b.properties_in1[t].flow_mass_comp[j]
+                 + b.properties_in2[t].flow_mass_comp[j]) ==
+                b.properties_treated[t].flow_mass_comp[j])
+
+    self._stream_table_dict = {"Inlet 1": self.inlet1,
+                               "Inlet 2": self.inlet2,
+                               "Treated": self.treated}
+
+    self._perf_var_dict["Solute Removal"] = self.removal_frac_mass_solute
+
+    self._get_Q = _get_Q_diso
+
+
+def initialize_diso(blk, state_args=None, outlvl=idaeslog.NOTSET,
+                    solver=None, optarg=None):
+    '''
+    Initialization routine for double inlet-single outlet unit models.
+
+    Keyword Arguments:
+        state_args : a dict of arguments to be passed to the property
+                       package(s) to provide an initial state for
+                       initialization (see documentation of the specific
+                       property package) (default = {}).
+        outlvl : sets output level of initialization routine
+        optarg : solver options dictionary object (default=None, use
+                 default solver options)
+        solver : str indicating which solver to use during
+                 initialization (default = None, use default IDAES solver)
+
+    Returns:
+        None
+    '''
+    if optarg is None:
+        optarg = {}
+
+    # Set solver options
+    init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+    solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+    solver_obj = get_solver(solver, optarg)
+
+    # Get initial guesses for inlet if none provided
+    if state_args is None:
+        state_args = {}
+        state_dict = (
+            blk.properties_in1[
+                blk.flowsheet().time.first()]
+                .define_port_members())
+
+        for k in state_dict.keys():
+            if state_dict[k].is_indexed():
+                state_args[k] = {}
+                for m in state_dict[k].keys():
+                    state_args[k][m] = state_dict[k][m].value
+            else:
+                state_args[k] = state_dict[k].value
+
+    # ---------------------------------------------------------------------
+    # Initialize state blocks
+    flags = blk.properties_in1.initialize(
+        outlvl=outlvl,
+        optarg=optarg,
+        solver=solver,
+        state_args=state_args,
+        hold_state=True
+    )
+    blk.properties_in2.initialize(
+        outlvl=outlvl,
+        optarg=optarg,
+        solver=solver,
+        state_args=state_args,
```

Comment:
I don't think we should be passing the same set of state args to inlet 2 as for inlet 1. Seeing as this should already be set, we could probably just not send any state args here.